app/forms/piv_cac_proofing_form.rb

```diff
@@ -0,0 +1,50 @@
+class PivCacProofingForm
+  include ActiveModel::Model
+  include PivCacFormHelpers
+
+  attr_accessor :x509_dn_uuid, :x509_dn, :token, :error_type, :nonce, :user, :key_id, :first_name,
+                :last_name, :cn
+
+  validates :token, presence: true
+  validates :nonce, presence: true
+
+  def submit
+    success = valid? && valid_submission?
+
+    errors = error_type ? { type: error_type } : {}
+    FormResponse.new(
+      success: success,
+      errors: errors,
+      extra: extra_analytics_attributes.merge(error_type ? { key_id: key_id } : {}),
+    )
+  end
+
+  def card_type
+    @data['card_type']
+  end
+
+  private
+
+  def valid_submission?
+    valid_token? && valid_full_name_or_cn?
+  end
+
+  def valid_full_name_or_cn?
+    return if @data.nil? || @data['card_type'].blank?
+    @cn = PivCac::ExtractCnFromSubject.call(@data['subject'])
+    return if cn.blank?
+    @last_name, @first_name = cn.split('.') if card_type == 'cac'
```

Comment:
can we get a unit test for this class? I'd love to see specs on properties like this being set correctly